src/internal/client.ts

```diff
@@ -1626,4 +1624,108 @@ export class TypedClient {
 
     await this.makeRequestAsyncOmit({ method, bucketName, query, headers }, payload)
   }
+
+  async setTagging(taggingParams: PutTaggingParams): Promise<void> {
+    const { bucketName, objectName, tags, putOpts } = taggingParams
+    const method = 'PUT'
+    let query = 'tagging'
+
+    if (putOpts && putOpts?.versionId) {
+      query = `${query}&versionId=${putOpts.versionId}`
+    }
+    const tagsList = []
+    for (const [key, value] of Object.entries(tags)) {
+      tagsList.push({ Key: key, Value: value })
+    }
+    const taggingConfig = {
+      Tagging: {
+        TagSet: {
+          Tag: tagsList,
+        },
+      },
+    }
+    const headers = {} as RequestHeaders
+    const builder = new xml2js.Builder({ headless: true, renderOpts: { pretty: false } })
+    const payloadBuf = Buffer.from(builder.buildObject(taggingConfig))
+    const requestOptions = {
+      method,
+      bucketName,
+      query,
+      headers,
+
+      ...(objectName && { objectName: objectName }),
+    }
+
+    headers['Content-MD5'] = toMd5(payloadBuf)
+
+    await this.makeRequestAsyncOmit(requestOptions, payloadBuf)
+  }
+
+  async removeTagging({ bucketName, objectName, removeOpts }: RemoveTaggingParams): Promise<void> {
```

Comment:
Should add `private` identifier to make sure this function is never called from outside of this class.

```suggestion
  private async removeTagging({ bucketName, objectName, removeOpts }: RemoveTaggingParams): Promise<void> {
```